webapp/src/components/issue_attribute_selector.jsx

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import ReactSelect from 'react-select';
+import PropTypes from 'prop-types';
+
+import {getStyleForReactSelect} from 'utils/styles';
+import Setting from 'components/setting';
+
+export default class IssueAttributeSelector extends PureComponent {
+    static propTypes = {
+        name: PropTypes.string,
+        repo: PropTypes.string,
+        required: PropTypes.bool,
+        isMulti: PropTypes.bool,
+        resetInvalidOnChange: PropTypes.bool,
+        addValidate: PropTypes.func,
+        removeValidate: PropTypes.func,
+        onChange: PropTypes.func,
+        load: PropTypes.func,
+        theme: PropTypes.object.isRequired,
+        value: PropTypes.oneOfType([
+            PropTypes.object,
+            PropTypes.array,
+            PropTypes.string,
+        ]),
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            options: [],
+            values: [],
+            isLoading: false,
+            invalid: false,
+            error: '',
+        };
+    }
+
+    componentDidMount() {
+        if (this.props.addValidate) {
+            this.props.addValidate(this.isValid);
+        }
+    }
+
+    componentWillUnmount() {
+        if (this.props.removeValidate) {
+            this.props.removeValidate(this.isValid);
+        }
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (prevState.invalid && this.props.value !== prevProps.value) {
+            this.setState({invalid: false}); //eslint-disable-line react/no-did-update-set-state
+        }
+
+        if (prevProps.repo !== this.props.repo) {
+            this.loadOptions();
+        }
+    }
+
+    loadOptions = async () => {
+        this.setState({isLoading: true});
+        const options = await this.props.load();
+
+        // filter out currently selected options that do not exist in the new repo
+        const optionValues = options.map((option) => option.value);
+        const validValues = this.state.values.filter((value) => optionValues.includes(value.value));
+
+        this.setState({
+            options,
+            values: validValues,
+            isLoading: false,
+        });
+    }
+
+    onChange = (values) => {
+        this.setState({values});
+
+        if (!values) {
+            this.props.onChange(this.props.isMulti ? [] : '');
+            return;
+        }
```

Comment:
This is usually an anti-pattern to have values in the parent's state and the child component's state. I recommend removing the `values` property from this component's state, and instead always relying on the parent's `value` prop as the source of truth.

Reply:
@mickmister This is a good point. I realized that the `value` prop and some other props were actually unused in this component, so I got rid of those. 

Now there's only one source of truth for the selected values and it's in this component's state. It's a bit simpler to keep this state here as opposed to moving it to the parent's state because otherwise, I'd have to define states for the upcoming assignee & milestone selector components as well.

Reply:
@utkuufuk What do you think about using the prop instead of storing the value in the child component's state? I feel very strongly about only storing the component's value in the parent. It's fine to store the options in the child component, as the parent doesn't need to know about the available choices, but the selected values should only be stored in one place.

Reply:
Currently the selected values are being stored in the parent's `labels` state, and the child component's `values` state. I propose only storing them in the `labels` state, and have the child component rely on the prop coming in. We want to have mostly stateless [controlled components](https://reactjs.org/docs/forms.html#controlled-components) whenever possible.

Reply:
@mickmister Ahh you're 100% correct. I completely missed the point that we're storing the labels in `CreateIssueModal` too :smile:

Reply:
@mickmister Done: https://github.com/mattermost/mattermost-plugin-github/pull/343/commits/3e3dc2537f9683d398dd6359b8d4f685bce03e2d